### PR TITLE
Mongo utils

### PR DESCRIPTION
**Pull request by ko-j**
- add mongodb utils
- update pom.xml files

### CODE REVIEW

**Comment by kaniyan** [comment]
@ko-j Can you please rebase it, there are some conflicting files.

File: services/device-registry/device-registry-mongodb/pom.xml [hunk 1]
```diff
@@ -67,7 +86,7 @@
                                             <port>5671</port>
                                             <port>5672</port>
                                             <port>8080</port>
-                                            <port>8443</port>
+                                            <port>8443</port>junitparams
```
Comment (kaniyan) [inline services/device-registry/device-registry-mongodb/pom.xml]: _junitparams_ here? Also please build and check.

File: services/device-registry/device-registry-mongodb/pom.xml [hunk 2]
```diff
@@ -1,11 +1,24 @@
...
+    SPDX-License-Identifier: EPL-2.0
+ -->
 <project xmlns="http://maven.apache.org/POM/4.0.0"
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <parent>
         <artifactId>hono-service-device-registry</artifactId>
         <groupId>org.eclipse.hono</groupId>
         <version>1.2.0-SNAPSHOT</version>
+        <relativePath>..</relativePath>
```
Comment (kaniyan) [inline services/device-registry/device-registry-mongodb/pom.xml]: @ko-j This relative path is not needed here. The build fails.

**Review by kaniyan — CHANGES_REQUESTED** [review]

File: services/device-registry/device-registry-mongodb/pom.xml (hunk 1 above)
Comment (ko-j) [reply services/device-registry/device-registry-mongodb/pom.xml]: resolved

File: services/device-registry/device-registry-mongodb/pom.xml (hunk 2 above)
Comment (ko-j) [reply services/device-registry/device-registry-mongodb/pom.xml]: resolved

File: services/device-registry/device-registry-mongodb/pom.xml [hunk 3]
```diff
@@ -25,6 +37,18 @@
...
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.12</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.12</version>
+            <scope>test</scope>
```
Comment (kaniyan) [inline services/device-registry/device-registry-mongodb/pom.xml:50]: These 2 junit dependencies are not being used and can be removed.

File: services/device-registry/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/utils/MongoDBCallExecutor.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    private static final Logger log = LoggerFactory.getLogger(MongoDBCallExecutor.class);
+
+    private Vertx vertx;
+    private MongoClient mongoClient;
+    private MongoDbConfigProperties mongoDbConfig;
+
+    /**
+     * construct executor to init mongodb properties.
+     *
+     * @param vertx         acting vertx context
```
Comment (kaniyan) [inline services/device-registry/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/utils/MongoDBCallExecutor.java:42]: How about `@param vertx The Vert.x instance to use.`

File: services/device-registry/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/utils/MongoDBCallExecutor.java [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+     * @param mongoDbConfig mongodb config properties
```
Comment (kaniyan) [inline services/device-registry/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/utils/MongoDBCallExecutor.java:43]: How about `The mongodb configuration properties to use.`?

**Review by kaniyan — CHANGES_REQUESTED** [review]

**Pull request merged**